src/wallet/wallet.h

```diff
@@ -919,7 +919,7 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     void BlockDisconnected(const std::shared_ptr<const CBlock>& pblock) override;
     bool AddToWalletIfInvolvingMe(const CTransactionRef& tx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate);
     int64_t RescanFromTime(int64_t startTime, bool update);
-    CBlockIndex* ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate = false);
+    CBlockIndex* ScanForWalletTransactions(CBlockIndex* pindexStart, CBlockIndex* pindexStop, bool fUpdate = false);
```

Comment:
`CBlockIndex* pindexStop = nullptr` will let existing code default to current behavior (mostly in the test cases in `wallet/test/wallet_tests.cpp`).

Reply:
Heh. I had it before but then @jnewbery convinced me to remove it: https://github.com/bitcoin/bitcoin/pull/7061#discussion_r138470085